core/src/main/kotlin/com/alecstrong/sql/psi/core/SqlCoreEnvironment.kt

```diff
@@ -75,24 +75,25 @@ open class SqlCoreEnvironment(
     val contributorIndex = CoreFileIndex(sourceFolders + dependencies, localFileSystem, projectEnvironment.project)
     projectEnvironment.project.picoContainer
         .registerComponentInstance(SchemaContributorIndex::class.java.name, object : SchemaContributorIndex {
+          private val contributors by lazy {
+            val manager = PsiManager.getInstance(projectEnvironment.project)
+            val map = linkedMapOf<VirtualFile, Collection<SchemaContributor>>()
+            contributorIndex.iterateContent { file ->
```

Comment:
Would this work when someone adds a new file with, say, a `CREATE INDEX` (which has a `SchemaContributor`) after the `contributors` field has been evaluated?

Reply:
`SqlCoreEnvironment` is the fake IntelliJ instance that runs during CLI/gradle builds, so in this case it'll be re-evaluated when they do an incremental compilation

Reply:
Gotcha, thanks!